### PR TITLE
Fix models scripts/notebooks

### LINKED ISSUE #1052: [BUG]: Some notebook & scripts in models don't run
### Version

23.07

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

   - A broken reference to clx.analytics.periodicity_detection in dfp-hamma inference script. 
   - ABP notebook metric score needs to change to numpy inorder to call `sklearn.metric` methods.
   - Invalid `x_val` parameter in the `AutoEncoder` model in Hammah notebook.

### Minimum reproducible example

_No response_

### Relevant log output

_No response_

### Full env printout

_No response_

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by tzemicheal**
## Description

- Update broken reference in the hamma_inference script
- Fix ABP notebook scoring of cupy/cudf output to numpy.
- Fix AutoEncoder parameters in Hammah notebook.
- Add peridicity_detection module for hamma inference.

Closes #1052 
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by hsin-c — APPROVED** [review]
looks good to me

**Review by raykallen — APPROVED** [review]
